custom_components/tuya_local/devices/eurom_sani_bathroom_towel_radiator.yaml

```diff
@@ -0,0 +1,152 @@
+name: Eurom Sani-Bathroom-Radiator Towel Wifi heater
+primary_entity:
+  entity: climate
+  translation_key: heater
+  dps:
+    - id: 1
+      type: boolean
+      name: hvac_mode
+      mapping:
+        - dps_val: false
+          value: "off"
+        - dps_val: true
+          constraint: mode
+          conditions:
+            - dps_val: "r"
+              value: "heat"
+            - dps_val: "t"
+              value: "dry"
+              icon: "mdi:hanger"
+    - id: 2
+      type: integer
+      name: temperature
+      range:
+        min: 0
+        max: 37
+      unit: C
+      mapping:
+        - constraint: mode
+          conditions:
+            - dps_val: "t"
+              value_redirect: towel_dry_temperature
+              range:
+                min: 40
+                max: 70
+    - id: 3
+      type: integer
+      name: current_temperature
+      unit: C
+      mapping:
+        - constraint: mode
+          conditions:
+            - dps_val: "t"
+              value_mirror: surface_temperature_internal
+#    - id: 333
+#      # dummy variable for current temperature of
+#      # current mode (room / towel) while keeping
+#      # both measured values available
+#      type: integer
+#      name: current_temperature
+#      optional: true
+#      # read_only: true
+#      unit: C
+#      mapping:
+#        - constraint: mode
+#          conditions:
+#            - dps_val: "t"
+#              value_mirror: surface_temperature_internal
+#            - dps_val: "r"
+#              value_mirror: room_temperature_internal
+#    - id: 3
+#      type: integer
+#      name: room_temperature_internal
+#      hidden: true
+    - id: 105
+      name: surface_temperature_internal
+      type: integer
+      optional: true
+      hidden: true
+    - id: 12
+      name: error
+      type: bitfield
+      mapping:
+        - dps_val: 0
+          value: "OK"
+    - id: 101
+      type: string
+      name: hvac_action
+      mapping:
+        - dps_val: true
+          constraint: mode
+          conditions:
+            - dps_val: "r"
+              value: "heating"
+            - dps_val: "t"
+              #value: "heating"
+              value: "drying"
+        - dps_val: false
+          value: idle
+          constraint: hvac_mode
+          conditions:
+            - dps_val: false
+              value: "off"
+    - id: 4
+      name: mode
+      type: string
+      mapping:
+        - dps_val: "r"
+          value: "room"
+        - dps_val: "t"
+          value: "towel"
+    - id: 106
+      name: towel_dry_temperature
+      type: integer
+      hidden: true
+      optional: true
+      unit: C
+      range:
+        min: 40
+        max: 70
+    - id: 103
+      name: preset_mode
+      type: boolean
+      mapping:
+        - dps_val: false
+          value: "none"
```

Comment:
Appropriate standard presets would be `home` and `away`

Reply:
> Appropriate standard presets would be `home` and `away`

That appears to be a bit of a problem. When one of the modes is on, to switch to the other, the first should first be switched off. So Eco to Anti-frost switching should first switch Eco off, then in a next all switch Anti-frost on. We can make both the switches boolean secondary entities, but they should still be mutually exclusive, and to function properly, the switching should be a 2-step process.

Reply:
If the device does not accept these being switched together, then we don't really have a way to enforce that. Perhaps you can make some buttons in the UI that are tied to automations that do the correct sequence of events. In that case you may want to move both out to external switches which can be hidden from direct use in the UI, to avoid someone accidentally using the climate presets while the eco switch is not in the correct state.

Reply:
I had hoped there would be another way, to implement all functions of the heater, elegantly in the standard Climate entity format, while using this integration. Apparently the Tuya app also sets those in succession. I guess there are more Tuya devices out there that suffer from the same behavior.
But I understand that it is not straightforward implementation, while maintaining the elegant condition scheme that implicitly sets multiple states according to the yaml file of the device. Maybe adding some statements that define an order of entities to be set, which then determines whether values/switches can be set at once, or need successive calls to the API? Or that define 'invalid combinations', such that the integration knows that two contradicting states are not valid, then queues the states that are not allowed in combinations with others, and iterates until all combinations are valid and set. So if 'invalid: eco=on & away=on' is set, then switching from 'eco=on&away=off' will first switch to the valid 'eco=off&away=off', notices that it's not ready since away should be on in the end, and makes a second call to 'eco=off&away=on'. No defined sequence needed, only defining invalid combinations.